File: sample-data-repository/src/test/java/nl/haploid/octowight/sample/repository/RoleDmoRepositoryIT.java
package nl.haploid.octowight.sample.repository;

import nl.haploid.octowight.sample.AbstractIT;
import nl.haploid.octowight.sample.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

import static org.junit.Assert.assertEquals;

public class RoleDmoRepositoryIT extends AbstractIT {

	@Autowired
	private PersonDmoRepository personDmoRepository;

	@Autowired
	private RoleDmoRepository roleDmoRepository;

	@Test
	@Transactional
	public void testFindAll() {
		final PersonDmo personDmo = personDmoRepository.saveAndFlush(TestData.personDmo());
		final RoleDmo roleDmo = TestData.roleDmo(personDmo, "cleaner");
		roleDmoRepository.saveAndFlush(roleDmo);
		final List<RoleDmo> roleDmos = roleDmoRepository.findAll();
		assertEquals(1, roleDmos.size());
		assertEquals(personDmo.getName(), roleDmos.get(0).getPerson().getName());
	}
}

File: sample-data-repository/src/test/scala/nl/haploid/octowight/sample/repository/RoleDmoRepositoryIT.scala
package nl.haploid.octowight.sample.repository

import nl.haploid.octowight.sample.{AbstractTransactionalIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class RoleDmoRepositoryIT extends AbstractTransactionalIT {
  @Autowired private[this] val personDmoRepository: PersonDmoRepository = null
  @Autowired private[this] val roleDmoRepository: RoleDmoRepository = null

  behavior of "Role repository"

  it should "find all" in {
    roleDmoRepository.deleteAll()
    val personDmo = personDmoRepository.saveAndFlush(TestData.personDmo)
    val roleDmo = TestData.roleDmo(personDmo, "cleaner")
    roleDmoRepository.saveAndFlush(roleDmo)
    val roleDmos = roleDmoRepository.findAll
    roleDmos should have size 1
    roleDmos.get(0).getPerson.getName should be(personDmo.getName)
  }
}
